bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/discovery/HaywardDiscoveryService.java

```diff
@@ -0,0 +1,332 @@
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Sets up the discovery results and details
+ *
+ * @author Matt Myers - Initial contribution
+ */
+
+@NonNullByDefault
+public class HaywardDiscoveryService extends AbstractDiscoveryService implements DiscoveryService, ThingHandlerService {
+    private final Logger logger = LoggerFactory.getLogger(HaywardDiscoveryService.class);
+    private @Nullable HaywardBridgeHandler discoveryBridgehandler;
+
+    public HaywardDiscoveryService() {
+        super(THING_TYPES_UIDS, 0, false);
+    }
+
+    /**
+     * Constructs a zone discovery service.
+     * Registers this zone discovery service programmatically.
+     * Call {@link ZoneDiscoveryService#destroy()} to unregister the service after use.
+     */
+    public HaywardDiscoveryService(HaywardBridgeHandler bridge) throws IllegalArgumentException {
+        super(THING_TYPES_UIDS, 0, false);
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        HaywardBridgeHandler bridgehandler = discoveryBridgehandler;
+        try {
+            if (bridgehandler != null) {
+                String xmlResults = bridgehandler.getMspConfig();
+                mspConfigDiscovery(xmlResults);
+            }
+        } catch (HaywardException e) {
+            logger.warn("Exception during discovery scan: {}", e.getMessage());
+        } catch (InterruptedException e) {
+            return;
+        }
+    }
+
+    public synchronized boolean mspConfigDiscovery(String xmlResponse) {
+        List<String> systemIDs = new ArrayList<>();
+        List<String> names = new ArrayList<>();
+        List<String> bowName = new ArrayList<>();
+        List<String> bowID = new ArrayList<>();
+        List<String> property1 = new ArrayList<>();
+        List<String> property2 = new ArrayList<>();
+        List<String> property3 = new ArrayList<>();
+        List<String> property4 = new ArrayList<>();
+        Map<String, Object> properties = new HashMap<>();
+
+        HaywardBridgeHandler bridgehandler = discoveryBridgehandler;
+
+        if (bridgehandler != null) {
+            // Find Backyard
+            names = bridgehandler.evaluateXPath("//Backyard/Name/text()", xmlResponse);
+
+            for (int i = 0; i < names.size(); i++) {
+                properties.clear();
+                properties.put(HaywardBindingConstants.PROPERTY_TYPE, HaywardTypeToRequest.BACKYARD);
+                properties.put(HaywardBindingConstants.PROPERTY_SYSTEM_ID, bridgehandler.account.mspSystemID);
+
+                onDeviceDiscovered(HaywardBindingConstants.THING_TYPE_BACKYARD, names.get(i), properties);
```

Comment:
If system id is your representation property for a backyard, then this code would only be able to discover a single backyard since all backyard things would be assigned the same system id.

Reply:
Correct.  There is only one backyard per MSP account.